### DISCUSSION TITLE
How to update user data returned from authenticator.isAuthenticated?

### DISCUSSION THREAD

**Discussion by siimaoo** [pbteja1998/remix-auth-email-link#17] [Q&A] [answered]
I have something like this: 
```typescript
authenticator.use(
    new EmailLinkStrategy(
        {
	    sendEmail,
	    secret,
	    callbackURL: '/auth',
	},
	async ({ email }: { email: string }) => {
	    const user = await getUserByEmail(email);
	    return user;
	},
    ),
);
```

But if I update user, it's always returning old data when I get it from authenticator.isAuthenticated.
How can I update user data and when I use `const user = await authenticator.isAuthenticated(request)` get the updated user data?

**Comment by pbteja1998** [comment, marked as answer]
@siimaoo you would need to also update the cookie when you update the user. Otherwise you would be reading the data from a cookie that has old data.

So, there needs to be a set cookie header happening in your update user action.
